### PR TITLE
New keyboard - keyprez unicorn

### PR DESCRIPTION
**Pull request by csandven**
Closed the other one, recloned my fork and recreated the PR. Sorry about the mess, seems like I had a really old version of qmk, and it was a little bit messed up.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keyprez/unicorn/config.h [hunk 1]
```diff
@@ -0,0 +1,149 @@
+// Copyright %(YEAR)s %(YOUR_NAME)s (@%(USER_NAME)s)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x6b7a
```
Comment (drashna) [inline keyboards/keyprez/unicorn/config.h:9]:
```suggestion
#define VENDOR_ID    0x6B7A
```

File: keyboards/keyprez/unicorn/config.h [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+//#define BOOTMAGIC_LITE_ROW 0
+//#define BOOTMAGIC_LITE_COLUMN 0
+
+#define ENCODERS_PAD_A { D4 }
+#define ENCODERS_PAD_B { D2 }
+#define ENCODER_RESOLUTIONS { 4 }
+#define ENCODERS_PAD_A_RIGHT { D4 }
+#define ENCODERS_PAD_B_RIGHT { D2 }
+#define ENCODER_RESOLUTIONS_RIGHT {4}
+#define ENCODER_DIRECTION_FLIP
```
Comment (drashna) [inline keyboards/keyprez/unicorn/config.h:149]:
```suggestion
#define ENCODERS_PAD_A { D2 }
#define ENCODERS_PAD_B { D4 }
#define ENCODER_RESOLUTIONS { 4 }
#define ENCODERS_PAD_A_RIGHT { D2 }
#define ENCODERS_PAD_B_RIGHT { D4 }
#define ENCODER_RESOLUTIONS_RIGHT {4}
```

File: keyboards/keyprez/unicorn/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+};
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (clockwise) {
+        tap_code_delay(KC_VOLU, 10);
+    } else {
+        tap_code_delay(KC_VOLD, 10);
+    }
+    return false;
+}
```
Comment (drashna) [inline keyboards/keyprez/unicorn/keymaps/default/keymap.c:39]:
```suggestion
```

File: keyboards/keyprez/unicorn/keymaps/jorge/keymap.c [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+};
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (clockwise) {
+        tap_code_delay(KC_VOLU, 10);
+    } else {
+        tap_code_delay(KC_VOLD, 10);
+    }
+    return false;
+}
```
Comment (drashna) [inline keyboards/keyprez/unicorn/keymaps/jorge/keymap.c:41]:
```suggestion
```

File: keyboards/keyprez/unicorn/unicorn.c [hunk 5]
```diff
@@ -0,0 +1,4 @@
+// Copyright 2022 Christian Sandven (@csandven)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "unicorn.h"
```
Comment (drashna) [inline keyboards/keyprez/unicorn/unicorn.c:4]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "unicorn.h"

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return true;
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keyprez/unicorn/readme.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# Unicorn
+
+*A short description of the keyboard/project*
```
Comment (drashna) [inline keyboards/keyprez/unicorn/readme.md:3]: Fill this out?

File: keyboards/keyprez/unicorn/config.h [hunk 7]
```diff
@@ -0,0 +1,148 @@
+// Copyright %(YEAR)s %(YOUR_NAME)s (@%(USER_NAME)s)
```
Comment (zvecr) [inline keyboards/keyprez/unicorn/config.h:1]: These need filling out in the various added files

File: keyboards/keyprez/unicorn/info.json [hunk 8]
```diff
@@ -0,0 +1,10 @@
+{
+    "keyboard_name": "Unicorn",
+    "url": "",
+    "maintainer": "Keyprez",
+    "layouts": {
+        "LAYOUT_ISO": {
```
Comment (zvecr) [inline keyboards/keyprez/unicorn/info.json:6]:
```suggestion
        "LAYOUT": {
```

`LAYOUT_iso` would also be accepted.

File: keyboards/keyprez/unicorn/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+    _BASE,
+    _FN
+};
+
+#define FN MO(_FN)
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [_BASE] = LAYOUT_ISO(
```
Comment (zvecr) [inline keyboards/keyprez/unicorn/keymaps/default/keymap.c:15]:
```suggestion
    [_BASE] = LAYOUT(
```

File: keyboards/keyprez/unicorn/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_HOME, KC_MUTE,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,            KC_END, KC_DEL,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_NUHS, KC_ENT, KC_PGUP, KC_INSERT,
+        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP, KC_PGDN, KC_COPY,
+        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  KC_SPC,                             KC_SPC,  KC_RALT, FN,      KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT, KC_PASTE
+    ),
+
+    [_FN] = LAYOUT_ISO(
```
Comment (zvecr) [inline keyboards/keyprez/unicorn/keymaps/default/keymap.c:23]:
```suggestion
    [_FN] = LAYOUT(
```

File: keyboards/keyprez/unicorn/unicorn.h [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+/* This is a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT_ISO( \
```
Comment (zvecr) [inline keyboards/keyprez/unicorn/unicorn.h:16]:
```suggestion
#define LAYOUT( \
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**